src/azure-firewall/azext_firewall/custom.py

```diff
@@ -929,4 +972,81 @@ def remove_azure_firewall_policy_filter_rule(cmd, resource_group_name, firewall_
+                                       source_ip_groups=_judge(source_ip_groups, rule.source_ip_groups),
+                                       destination_ip_groups=_judge(destination_ip_groups, rule.destination_ip_groups))
+            elif rule.rule_type == 'ApplicationRule':
+                def map_application_rule_protocol(item):
+                    return FirewallPolicyRuleApplicationProtocol(protocol_type=item['protocol_type'],
+                                                                 port=int(item['port']))
+
+                protocols = list(map(map_application_rule_protocol, protocols))
+                new_rule = ApplicationRule(name=rule_name,
+                                           description=_judge(description, rule.description),
+                                           rule_type=rule.rule_type,
+                                           source_addresses=_judge(source_addresses, rule.source_addresses),
+                                           protocols=_judge(protocols, rule.protocols),
+                                           destination_addresses=_judge(destination_addresses, rule.destination_addresses),
+                                           fqdn_tags=_judge(fqdn_tags, rule.fqdn_tags),
+                                           target_fqdns=_judge(target_fqdns, rule.target_fqdns),
+                                           target_urls=_judge(target_urls, rule.target_urls),
+                                           source_ip_groups=_judge(source_ip_groups, rule.source_ip_groups),
+                                           terminate_tls=_judge(enable_tls_inspection, rule.terminate_tls),
+                                           web_categories=_judge(web_categories, rule.web_categories))
+            elif rule.rule_type == 'NatRule':
+                new_rule = NatRule(name=rule_name,
+                                   description=_judge(description, rule.description),
+                                   rule_type=rule.rule_type,
+                                   ip_protocols=_judge(ip_protocols, rule.ip_protocols),
+                                   source_addresses=_judge(source_addresses, rule.source_addresses),
+                                   destination_addresses=_judge(destination_addresses, rule.destination_addresses),
+                                   destination_ports=_judge(destination_ports, rule.destination_ports),
+                                   translated_address=_judge(translated_address, rule.translated_address),
+                                   translated_port=_judge(translated_port, rule.translated_port),
+                                   source_ip_groups=_judge(source_ip_groups, rule.source_ip_groups))
+            if new_rule:
+                target_rule_collection.rules[i] = copy.deepcopy(new_rule)
+                return instance
+            else:
+                raise CLIError(f'Undefined rule_type : {rule.rule_type}')
```

Comment:
Please don't raise CLIError. You should use Errors defined in this [file](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py)